definitions/tools/star_align.cwl

```diff
@@ -0,0 +1,190 @@
+            position: 10
+            prefix: '--alignMatesGapMax'
+        doc: 'maximum gap between two mates, if 0, max intron gap will be determined by (2^winBinNbits)*winAnchorDistNbins
+        '
+    outSAMstrandField:
+        type: string
+        default: intronMotif
+        inputBinding:
+            position: 11
+            prefix: '--outSAMstrandField'
+        doc: |
+          None
+          string: Cufflinks-like strand field flag
+          None        ... not used
+          intronMotif ... strand derived from the intron motif. Reads with inconsistent and/or non-canonical introns are filtered out.
+          NOTE you need this option to be set yto intronmotif if for stringtie quantification
+    twopassMode:
+        type: string?
+        inputBinding:
+            position: 12
+            prefix: '--twopassMode'
+        doc: |
+           None
+           string: 2-pass mapping mode.
+           None        ... 1-pass mapping
+           Basic ... basic 2-pass mapping, with all 1st pass junctions inserted into the genome indices on the fly
+    outFilterMultimapNmax:
+        type: int?
+        inputBinding:
+            position: 13
+            prefix: '--outFilterMultimapNmax'
+        doc: 'int: read alignments will be output only if the read maps fewer than this value,otherwise no alignments will be output
+        '
+    sjdbOverhang:
+        type: string?
+        inputBinding:
+            position: 14
+            prefix: '--sjdbOverhang'
+        doc: 'int>0: length of the donor/acceptor sequence on each side of the junctions, ideally = (mate_length - 1)
+        '
+    gtf_file:
+        type: File?
+        inputBinding:
+            position: 15
+            prefix: '--sjdbGTFfile'
+        doc: |
+          string: path to the GTF file with annotations
+    outFileNamePrefix:
+        type: string
+        inputBinding:
+            position: 16
+            prefix: '--outFileNamePrefix'
+        doc: ' output files name prefix (including full or relative path). Can
+          only be defined on the command line 
+          '
+    readFilesCommand:
+        type: string?
+        inputBinding:
+            position: 17
+            prefix: '--readFilesCommand'
+        doc: ' command line to execute for each of the input file. This command should generate FASTA or FASTQ text and send it to stdout, zcat - to uncompress .gz files, bzcat - to uncompress .bz2 files, etc 
+        '
+    outSAMattributes:
+        type: string[]
+        default: ["NH", "HI", "AS", "NM", "MD"]
+        inputBinding:
+            position: 18
+            prefix: '--outSAMattributes'
+        doc: |
+          Standard
+          string: a string of desired SAM attributes, in the order desired for the output SAM
+          NH HI AS nM NM MD jM jI XS ... any combination in any order
+          Standard   ... NH HI AS nM
+          All        ... NH HI AS nM NM MD jM jI
+          None ... no attributes
+    outSAMattrRGline:
```

Comment:
Looking at the sample yaml you updated, it looks like this is intended to take in an array of arrays of strings, but this doesn't match up with the type here. Additionally, according to the documentation, each RG line should be separated by " , " which isn't the current behavior.  CWL has an itemSeparator field that will allow us to insert " , " between array elements, but I can't get that working with nested arrays in my own tests. Maybe this should just be an array of strings, where each string is the full RG line- in which case the yaml should be updated.